src/app/dataset-view/dataset.component.ts

```diff
@@ -63,9 +77,34 @@ export class DatasetComponent extends BaseDatasetDataComponent implements OnInit
             _.isNil(this.datasetBasics) ||
             this.datasetBasics.name !== urlDatasetInfo.datasetName ||
             this.datasetBasics.owner.accountName !== urlDatasetInfo.accountName ||
-            this.datasetViewType === DatasetViewTypeEnum.Flows
+            [DatasetViewTypeEnum.Overview, DatasetViewTypeEnum.Metadata].includes(this.datasetViewType)
         ) {
-            this.requestMainData(urlDatasetInfo);
+            this.datasetService
+                .requestDatasetHashLastBlock(urlDatasetInfo.accountName, urlDatasetInfo.datasetName)
+                .pipe(
+                    map((hash: string) => {
+                        if (hash !== this.currentHashLastBlock && this.datasetBasics) {
+                            this.updateCache(this.datasetBasics);
+                        }
+                    }),
+                    switchMap(() => this.datasetService.requestDatasetMainData(urlDatasetInfo)),
+                    tap(() => {
+                        this.mainDatasetQueryComplete$.next(urlDatasetInfo);
+                    }),
```

Comment:
Isn't it enough just to clean the cache?

Reply:
No, need to make a main request too